### PR TITLE
xemu.c: Window Adjusts to Display Size

### PR DESCRIPTION
**Pull request by MasonT8198**
This change is because without it there is an issue where the menu settings disappear because of a smaller screens

### CODE REVIEW

**Comment by GXTX** [comment]
Are people using 800x600 these days? Looking at Steams hardware survey (although a bit biased towards "gamers") 1024x768 is 0.2% of the population, `statcounter.com` puts the same at 1% globally.

**Comment by MasonT8198** [comment]
> Are people using 800x600 these days? Looking at Steams hardware survey (although a bit biased towards "gamers") 1024x768 is 0.2% of the population, `statcounter.com` puts the same at 1% globally.

It's the way it starts, on some laptop screens and some tv screens it'll start without the hud, and you have to fiddle around with it to make the hud reappear.

**Comment by medievil1** [comment]
experience this myself.. 1920x1080 here but my dpi on a 50" display is 150%
dunno it is related as well... without setting it to use application size in the override(win 10/file properties) it will start top and bottom touching the full 1080 res and with the top cut off.....
click the side to resize and boom, snaps into correct range

**Comment by MasonT8198** [comment]
> experience this myself.. 1920x1080 here but my dpi on a 50" display is 150%
> 
> dunno it is related as well... without setting it to use application size in the override(win 10/file properties) it will start top and bottom touching the full 1080 res and with the top cut off.....
> 
> click the side to resize and boom, snaps into correct range

It has to do with the way windows app scaling is. I have it happen on my 4k tv as well, it gets super big so it breaks xemu. Because scaling issues.

**Comment by mborgerson** [comment]
The preferred fix is to query maximum window size (size of screen) and create a window no larger than current screen

**Comment by MasonT8198** [comment]
> The preferred fix is to query maximum window size (size of screen) and create a window no larger than current screen

how would you go about doing this

**Comment by mborgerson** [comment]
See https://wiki.libsdl.org/SDL_GetCurrentDisplayMode

**Review by mborgerson — CHANGES_REQUESTED** [review]
Please check host resolution first as suggested

File: ui/xemu.c [hunk 1]
```diff
@@ -882,6 +882,13 @@ static void sdl2_display_init(DisplayState *ds, DisplayOptions *o)
     int i;
     SDL_SysWMinfo info;
 
+    SDL_DisplayMode current;
+    // Query Window Size and Print The Display Mode
```
Comment (mborgerson) [inline ui/xemu.c:886]: You have queried the screen size, now we need to make sure that we don't create a window larger than that size.

Comment (MasonT8198) [reply ui/xemu.c:886]: Yes, my trouble is trying to find the command to limit those bounds, in SDL there is a command to adjust the screen size to those bounds but not limit them @mborgerson

Comment (mborgerson) [reply ui/xemu.c:886]: You can limit them when we create the window where we specify window dimensions

**Comment by mborgerson** [comment]
This is unfinished. I'll close for now and we can return with a follow up PR.

**Pull request closed without merging**